ci: check GHA results in TeamCity

rail opened the pull request:
Bazel Essential CI is a no-op on master and release-24.1 since the GitHub Actions CI migration completed.

This PR adds a new TeamCity build configuration which polls the required GitHub action results and reflects the overall status in TeamCity.

Fixes: DEVINF-1124
Release note: None

rail committed:
ci: check GHA results in TeamCity

Bazel Essential CI is a no-op on master and release-24.1 since the GitHub
Actions CI migration completed.

This PR adds a new TeamCity build configuration which polls the required
GitHub action results and reflects the overall status in TeamCity.

Fixes: DEVINF-1124
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123421)
<!-- Reviewable:end -->

rail commented:
I added a new TC configuration and added it to the MPB deps.
Examples https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_CheckGhaResults?branch=123421&buildTypeTab=overview&mode=builds

rickystewart commented on pkg/cmd/github-action-poller/main.go:
```diff
@@ -0,0 +1,155 @@
+		statuses[c] = nil
+	}
+
+	client := github.NewClient(nil)
```
This is unauthenticated, right? We'll hit rate limits even more quickly than usual. Maybe it doesn't make a difference if we do few enough requests.

rail replied:
Correct. I can easily add auth if we want. Actually, it was there in the initial version. :)

rail commented:
bors r=rickystewart

rail commented:
bors r=rickystewart
